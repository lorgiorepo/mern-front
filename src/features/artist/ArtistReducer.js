import {FETCH_ARTISTS} from './actionTypes';

const initialState = {
  artists: [],
};

export default (state = initialState, action) => {
  console.log('action.type',action.type);
  switch (action.type) {
    case `${FETCH_ARTISTS}_FULFILLED`:
      return {...state, artists: action.payload.data};

    default:
      return state;
  }
};
