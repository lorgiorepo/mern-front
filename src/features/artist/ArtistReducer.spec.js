import {FETCH_ARTISTS} from './actionTypes';
import ArtistReducer from './ArtistReducer';

describe('features > random > ArtistReducer', () => {
    it('returns initial state, if non matched action is dispatched', () => {
        const initialState = {
            artists: [],
        };
    
        const action = {
            type: 'FOO',
        };
    
        expect(ArtistReducer(initialState, action)).toBe(initialState);
    });

    it('updates state according to dispatched GET_ARTISTS', () => {
        const initialState = {
            artists: [],
        };

        const payload = [
            {
                "_id": "5f9c4ea4d2a29b79b4b19e51",
                "name": "Lorgio",
                "lastname": "Trinidad",
                "nationality": "Peruvian",
                "about": "i'm dad",
                "createdAt": "2020-10-30T17:34:28.655Z",
                "updatedAt": "2020-10-30T17:34:28.655Z",
                "__v": 0
            }
        ];

        const action = {
            type: FETCH_ARTISTS,
            payload,
        };

        expect(ArtistReducer(initialState, action)).toMatchSnapshot();
    });
});