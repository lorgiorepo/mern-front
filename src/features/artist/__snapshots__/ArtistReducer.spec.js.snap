// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features > random > ArtistReducer updates state according to dispatched GET_ARTISTS 1`] = `
Object {
  "artists": Array [],
  "value": undefined,
}
`;
